runtime/pack/dist/opt/termdebug/plugin/termdebug.vim

```diff
@@ -1042,19 +1042,21 @@ endfunc
 " (newlines and surrounding whitespace)
 func s:CleanupExpr(expr)
   " replace all embedded newlines/tabs/...
-  let expr = substitute( a:expr, '\s', ' ', 'g')
+  let expr = substitute(a:expr, '\n', ' ', 'g')
+  let expr = substitute(expr, '\s+', ' ', 'g')
```

Comment:
Instead of this:

    let expr = substitute(a:expr, '\n', ' ', 'g')
    let expr = substitute(expr, '\s+', ' ', 'g')

Try this:

    let expr = substitute(a:expr, '\_s\+', ' ', 'g')

That's assuming `+` is a quantifier.  If it's meant to match a literal plus, try this:

    let expr = substitute(a:expr, '\_s+', ' ', 'g')

Reply:
Can you please elaborate if/how `\_s` matches not only space and tabs but also newlines? Did I miss something in `help pattern`?

Reply:
Rechecked: doesn't replace newlines (but of course the `+` still should be escaped).

Reply:
For every atom matching a class of character, like `\s`, `\d`, `\w`, ... there exists an extra atom (with an underscore after the backslash) matching the exact same characters plus the newline character:

    \s = [ \t]
    \_s = [ \t\n]

    \d = [0-9]
    \_d = [0-9\n]

    \w = [a-z]
    \_w = [a-z\n]

For more info, see [`:help \_x`](https://vimhelp.org/pattern.txt.html#%2F%5C_x)

Reply:
Interestingly the following works:

```viml
  let expr = substitute(a:expr, '\n', ' ', 'g')
  let expr = substitute(expr, '\s\+', ' ', 'g')
```

But that doesn't:


```viml
  let expr = substitute(a:expr, '\_s\+', ' ', 'g')
```

Maybe that doesn't apply to `substitute`?

Reply:
> Rechecked: doesn't replace newlines (but of course the + still should be escaped).

It works for me:
```vim
function Func(expr)
    let expr = substitute(a:expr, '\_s\+', ' ', 'g')
    echomsg expr
endfunction
call Func("aaa\n\n\nbbb   ccc")
```
    aaa bbb ccc

Notice that the 3 consecutive newlines have been replaced with a single space.

Reply:
I've obviously not tested what I've thought. Rechecked, works fine.